### DISCUSSION TITLE
Error on copy a page from one sharepoint site to another

### DISCUSSION THREAD

**Discussion by verdura** [pnp/powershell#319] [Q&A] [answered]
Hi all,

I work at a school and have a script to create sites, users and folders automatically. In the finaI steps of the script I need to copy a page from a Sharepoint to another and make it the homepage, but I can't and have been stuck...

Here's what I have so far...

```
$cred = Get-Credential
 Connect-PnPOnline -Url https://acme.sharepoint.com/sites/sitewiththepagemodel -Credentials  $cred
    
 $tempFile = [System.IO.Path]::GetTempFileName();
        
 Export-PnPPage -Identity "SitePages/Bem-vindo-ao-site-da-Turma.aspx" -Force -Out $tempFile
    
 Connect-PnPOnline -Url "https://acme.sharepoint.com/sites/sitebeingcreated" -Credentials  $cred
 Apply-PnPProvisioningTemplate -Path $tempFile
    
 Set-PnPHomePage -RootFolderRelativeUrl SitePages/Bem-vindo-ao-site-da-Turma.aspx
```
When run I get the error:

```
Export-PnPPage : The object reference was not defined as an instance of an object
 At line:630 char:1
 + Export-PnPPage -Identity "SitePages/Bem-vindo-ao-site-da-Turma.aspx"  ...
 + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     + CategoryInfo          : NotSpecified: (:) [Export-PnPPage], NullReferenceException
     + FullyQualifiedErrorId : System.NullReferenceException,PnP.PowerShell.Commands.Provisioning.Tenant.ExportPage
```
In the past I've used "Export-PnPClientSidePage" instead of "Export-PnPPage" and It worked. But in the last time it says:

`WARNING: Please use 'Export-PnPPage'. The alias 'Export-PnPClientSidePage' will be removed in the 1.5.0 release `

So I change the command but it doesn't work. What am I doing wrong?
I'm not very good at Powershell so please, be gentle!! ;)

Thanks a lot for your help

**Comment by gautamdsheth** [comment]
Can you run the `Get-PnPException` command immediately after you get this error ? You paste the stacktrace here, this will help us investigate the issue. Thanks !

**Reply by verdura** [reply]
Here it is:

```
Message          : A referência de objecto não foi definida como uma instância de um objecto.
Stacktrace       :    em PnP.Framework.Provisioning.ObjectHandlers.Utilities.ClientSidePageContentsHelper.ExtractClientSidePage(Web web, ProvisioningTemplate template, 
                   ProvisioningTemplateCreationInformation creationInfo, PnPMonitoredScope scope, PageToExport page)
                      em PnP.PowerShell.Commands.Provisioning.Tenant.ExportPage.ExtractTemplate(String dirName, String fileName, ExtractConfiguration configuration)
                      em PnP.PowerShell.Commands.Provisioning.Tenant.ExportPage.ProcessRecord()
                      em System.Management.Automation.CommandProcessor.ProcessRecord()
ScriptLineNumber : 6
```
Line 6 is the line of `Export-PnPPage`

**Reply by verdura** [reply]
I've uninstalled and Installed the module and now the error is that `Apply-PnPProvisioningTemplate` doesn't exist as a command...

**Comment by veronicageek** [comment, marked as answer]
The command has been renamed to `Invoke-PnPSiteTemplate`.
See [CHANGELOG.md](https://github.com/pnp/powershell/blob/dev/CHANGELOG.md) to keep up to date with any changes/improvements that we make to the module.
